src/file.ts

```diff
@@ -1536,31 +1536,16 @@ class File extends ServiceObject<File> {
           return;
         }
 
-        // TODO(https://github.com/googleapis/nodejs-storage/issues/709):
-        // Remove once the backend issue is fixed.
-        // If object is stored compressed (having
-        // metadata.contentEncoding === 'gzip') and was served decompressed,
-        // then skip checksum validation because the remote checksum is computed
-        // against the compressed version of the object.
-        if (!isServedCompressed) {
-          try {
-            await this.getMetadata({userProject: options.userProject});
-          } catch (e) {
-            throughStream.destroy(e as Error);
-            return;
-          }
-          if (this.metadata.contentEncoding === 'gzip') {
-            throughStream.end();
-            return;
-          }
-        }
+        // Did the server decompress object before serving it back to us?
+        const serverDecompressed =
+          headers['x-goog-stored-content-encoding'] === 'gzip' && !isCompressed;
 
         // If we're doing validation, assume the worst-- a data integrity
         // mismatch. If not, these tests won't be performed, and we can assume
         // the best.
         let failed = crc32c || md5;
 
-        if (validateStream) {
+        if (validateStream && !serverDecompressed) {
```

Comment:
Would add a comment detailing why we care about `!serverDecompressed`:

> We must check if the server decompressed the data on serve because hash validation is not possible in this case.